packages/core/src/app/shipping/MultiShippingFormV2.tsx

```diff
@@ -0,0 +1,109 @@
+import { Consignment } from '@bigcommerce/checkout-sdk';
+import React, { FunctionComponent, useMemo, useState } from 'react';
+
+import { withLanguage, WithLanguageProps } from '@bigcommerce/checkout/locale';
+import { useCheckout } from '@bigcommerce/checkout/payment-integration-api';
+
+import { withFormikExtended } from '../common/form';
+import { EMPTY_ARRAY } from '../common/utility';
+import { Button, ButtonVariant } from '../ui/button';
+
+import ConsignmentListItem from './ConsignmentListItem';
+import hasSelectedShippingOptions from './hasSelectedShippingOptions';
+import MultiShippingFormV2Footer from './MultiShippingFormV2Footer';
+import './MultiShippingFormV2.scss';
+import NewConsignment from './NewConsignment';
+
+interface MultiShippingFormV2Values {
+    orderComment: string;
+}
+
+export interface MultiShippingFormV2Props {
+    customerMessage: string;
+    defaultCountryCode?: string;
+    countriesWithAutocomplete: string[];
+    isLoading: boolean;
+    onUnhandledError(error: Error): void;
+    onSubmit(values: MultiShippingFormV2Values): void;
+}
+
+const MultiShippingFormV2: FunctionComponent<MultiShippingFormV2Props> = ({
+    countriesWithAutocomplete,
+    defaultCountryCode,
+    isLoading,
+    onUnhandledError,
+}: MultiShippingFormV2Props) => {
+    const [addShippingDestination, setAddShippingDestination] = useState(false);
```

Comment:
Does it make sense to change the state name to `isAddingShippingDestination`? I feel this is more in-line with [our existing naming usages](https://github.com/search?q=repo%3Abigcommerce%2Fcheckout-js+isadding&type=code).